### PR TITLE
Fix example EDS datasets

### LINKED ISSUE #2429: Example EDS error when saving as .hspy
Both the TEM and SEM EDS datasets raise an error when saving them to the hyperspy hdf5 format.
```python
s = hs.datasets.example_signals.EDS_TEM_Spectrum()
s.save('eds.hspy')
```
<details><summary>Error message</summary>
<p>

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
c:\users\thomasaar\documents\github\hyperspy\hyperspy\io_plugins\hspy.py in overwrite_dataset(group, data, key, signal_axes, chunks, **kwds)
    570             # contains the chunk shape guessed by h5py
--> 571             dset = group.require_dataset(key, **these_kwds)
    572             got_data = True

~\.conda\envs\py5\lib\site-packages\h5py\_hl\group.py in require_dataset(self, name, shape, dtype, exact, **kwds)
    190             if not name in self:
--> 191                 return self.create_dataset(name, *(shape, dtype), **kwds)
    192 

~\.conda\envs\py5\lib\site-packages\h5py\_hl\group.py in create_dataset(self, name, shape, dtype, data, **kwds)
    135         with phil:
--> 136             dsid = dataset.make_new_dset(self, shape, dtype, data, **kwds)
    137             dset = dataset.Dataset(dsid)

~\.conda\envs\py5\lib\site-packages\h5py\_hl\dataset.py in make_new_dset(parent, shape, dtype, data, chunks, compression, shuffle, fletcher32, maxshape, compression_opts, fillvalue, scaleoffset, track_times, external, track_order, dcpl)
    117             dtype = numpy.dtype(dtype)
--> 118         tid = h5t.py_create(dtype, logical=1)
    119 

h5py\h5t.pyx in h5py.h5t.py_create()

h5py\h5t.pyx in h5py.h5t.py_create()

h5py\h5t.pyx in h5py.h5t.py_create()

TypeError: No conversion path for dtype: dtype('<U2')

During handling of the above exception, another exception occurred:

KeyError                                  Traceback (most recent call last)
<ipython-input-7-56c3607e5bfd> in <module>
      1 s = hs.datasets.example_signals.EDS_TEM_Spectrum()
----> 2 s.save('eds.hspy')

c:\users\thomasaar\documents\github\hyperspy\hyperspy\signal.py in save(self, filename, overwrite, extension, **kwds)
   2229             basename, ext = os.path.splitext(filename)
   2230             filename = basename + '.' + extension
-> 2231         io.save(filename, self, overwrite=overwrite, **kwds)
   2232 
   2233     def _replot(self):

c:\users\thomasaar\documents\github\hyperspy\hyperspy\io.py in save(filename, signal, overwrite, **kwds)
    602                              "False.")
    603         if write:
--> 604             writer.file_writer(filename, signal, **kwds)
    605             _logger.info('The %s file was created' % filename)
    606             folder, filename = os.path.split(os.path.abspath(filename))

c:\users\thomasaar\documents\github\hyperspy\hyperspy\io_plugins\hspy.py in file_writer(filename, signal, *args, **kwds)
    753             smd.record_by = ""
    754         try:
--> 755             write_signal(signal, expg, **kwds)
    756         except BaseException:
    757             raise

c:\users\thomasaar\documents\github\hyperspy\hyperspy\io_plugins\hspy.py in write_signal(signal, group, **kwds)
    708         metadata_dict["_internal_parameters"] = \
    709             metadata_dict.pop("_HyperSpy")
--> 710     dict2hdfgroup(metadata_dict, mapped_par, **kwds)
    711     original_par = group.create_group(original_metadata)
    712     dict2hdfgroup(signal.original_metadata.as_dictionary(), original_par,

c:\users\thomasaar\documents\github\hyperspy\hyperspy\io_plugins\hspy.py in dict2hdfgroup(dictionary, group, **kwds)
    441     for key, value in dictionary.items():
    442         if isinstance(value, dict):
--> 443             dict2hdfgroup(value, group.create_group(key),
    444                           **kwds)
    445         elif isinstance(value, DictionaryTreeBrowser):

c:\users\thomasaar\documents\github\hyperspy\hyperspy\io_plugins\hspy.py in dict2hdfgroup(dictionary, group, **kwds)
    451             write_signal(value, group.require_group(kn))
    452         elif isinstance(value, (np.ndarray, h5py.Dataset, da.Array)):
--> 453             overwrite_dataset(group, value, key, **kwds)
    454         elif value is None:
    455             group.attrs[key] = '_None_'

c:\users\thomasaar\documents\github\hyperspy\hyperspy\io_plugins\hspy.py in overwrite_dataset(group, data, key, signal_axes, chunks, **kwds)
    574             # if the shape or dtype/etc do not match,
    575             # we delete the old one and create new in the next loop run
--> 576             del group[key]
    577     if dset == data:
    578         # just a reference to already created thing

h5py\_objects.pyx in h5py._objects.with_phil.wrapper()

h5py\_objects.pyx in h5py._objects.with_phil.wrapper()

~\.conda\envs\py5\lib\site-packages\h5py\_hl\group.py in __delitem__(self, name)
    397     def __delitem__(self, name):
    398         """ Delete (unlink) an item from this group. """
--> 399         self.id.unlink(self._e(name))
    400 
    401     @with_phil

h5py\_objects.pyx in h5py._objects.with_phil.wrapper()

h5py\_objects.pyx in h5py._objects.with_phil.wrapper()

h5py\h5g.pyx in h5py.h5g.GroupID.unlink()

KeyError: "Couldn't delete link (callback link pointer is NULL (specified link may be '.' or not exist))"```
```
</p>
</details>

The following works fine, which implies that there is something going on specifically with these datasets.

```python
s = hs.signals.Signal1D([1,2,3])
s.set_signal_type('EDS_SEM')
s.save('eds.hspy')
```

### PR DESCRIPTION
**Pull request by tjof2**
### Description of the change

- Change elements from np.array to list, closes #2429

Closes #2429 by replacing `s.metadata.Sample.elements` in the example signals with Python lists. I'm not totally sure why the file size changes by ~1KB, but the metadata is identical (apart from `s.metadata.Sample.elements` of course) as is `s.data`.

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] ~update docstring (if appropriate),~
- [x] ~update user guide (if appropriate),~
- [x] ~add tests,~
- [x] ready for review.

### Minimal example of the bug fix or the new feature

```python
>>> import hyperspy.api as hs
>>> s = hs.datasets.example_signals.EDS_TEM_Spectrum()
>>> s.save('eds.hspy') # no longer raises an error like before
```

### CODE REVIEW

**Comment by tjof2** [comment]
See #2479 for the unrelated Travis test failures

File: hyperspy/tests/io/test_hdf5.py [hunk 1]
```diff
@@ -654,7 +654,7 @@ def test_load_missing_y2_value(self):
 
 def test_strings_from_py2():
     s = EDS_TEM_Spectrum()
-    assert s.metadata.Sample.elements.dtype.char == "U"
+    assert isinstance(s.metadata.Sample.elements, list)
```
Comment (tjof2) [inline hyperspy/tests/io/test_hdf5.py:657]: Could just remove the test altogether, happy either way.

**Comment by tjof2** [comment]
Travis failure is due to docs - an issue with `nexus.py` docstrings somewhere - @ericpre

**Comment by ericpre** [comment]
If you are updating the files, is it worth using the `hspy` extension?

**Comment by tjof2** [comment]
> If you are updating the files, is it worth using the hspy extension?

Good idea. I'm actually away from a PC for the next 10 days, so I'll update it then!

**Comment by tjof2** [comment]
> If you are updating the files, is it worth using the `hspy` extension?

This is done.

**Pull request merged**